benchmark/batch_hessian.jl

```diff
@@ -0,0 +1,87 @@
+using LinearAlgebra
+using SparseArrays
+using KernelAbstractions
+using BenchmarkTools
+
+using ExaPF
+import ExaPF: LinearSolvers
+
+const LS = LinearSolvers
+
+function build_batch_nlp(datafile, device, nbatch)
+    print("Load data\t")
+    polar = @time PolarForm(datafile, device)
+
+    constraints = Function[
+        ExaPF.voltage_magnitude_constraints,
+        ExaPF.active_power_constraints,
+        ExaPF.reactive_power_constraints,
+    ]
+    powerflow_solver = NewtonRaphson(tol=1e-10)
+
+    J = ExaPF.jacobian_sparsity(polar, ExaPF.power_balance, State())
+    if isa(device, CPU)
+        lufac = lu(J)
+        linear_solver = LS.DirectSolver(lufac)
+        if nbatch == 1
+            hessian_lagrangian = ExaPF.HessianLagrangian(polar, lufac, lufac')
+        else
+            hessian_lagrangian = ExaPF.BatchHessianLagrangian(polar, lufac, lufac', nbatch)
```

Comment:
Isn't `HessianLagrangian` the same as `BatchHessianLagragian` with `nbatch=1? Why do we have to make it a special case?

Reply:
Good question. In fact, I think defining two separate structures (`BatchHessianLagrangian` and `HessianLagrangian`) is making sense, as 
1. it allows to use multiple dispatch to dispatch either using batch or single Hessian-vector products to compute the reduced Hessian
2. `HessianLagrangian` is storing `Vector` in its attributes, whereas `BatchHessianLagrangian` is storing `Matrix`